### ISSUE TITLE
Prepare code to allow STOCSY in multispectra 

### ISSUE THREAD

**Issue by lpatiny** [cheminfo/nmrium#1871]
Calculate the median / Q1 / Q3 / min / max

Be able to display median, Q1/Q3 and min/max

Color the median by STOCSY

All those options should be available in the preferences of multi spectra analysis

**Comment by lpatiny** [comment]
Sorry this was some notes during the meeting and the explanation was pretty short.

This issue is actually quite complex to implement and will require a lot of calculations / options.

We first need to generate for the spectra a matrix and for this we need to implement those 2 features:

- https://github.com/cheminfo/nmrium/issues/1892
- https://github.com/cheminfo/nmrium/issues/1893

Once this is done all the 'x' array of all the spectra should be the same (same from / to and same length).

You can then call the method `matrixBoxPlot` from ml-spectra-processing (https://github.com/mljs/spectra-processing/issues/169)

We can then based on the preferences draw 2 areas and a median chart.

<img width="639" alt="image" src="https://user-images.githubusercontent.com/1484241/201069966-96c3ecab-66a8-4448-8443-c2c6fc533251.png">

The color of the median is still another issue



will be able to calculate using https://github.com/mljs/spectra-processing/blob/master/src/x/xBoxPlot.ts 6 arrays:

**Comment by lpatiny** [comment]
![image](https://github.com/cheminfo/nmrium/assets/1484241/82a53a19-a09d-4830-89a1-47804bcdf732)

This should be developed behind the experimental flag.

4 buttons:
- Hide all spectra (active if at least one is visible)
- Show all spectra (active if at least one is not visible)
- Show BoxPlot (toggle button)
- Show STOCSY (toggle button)


We need to calculate derived data from the matrix.
```
function boxPlot(matrix) {
  return {
    min: [],
    max: [],
    median: [],
    q1: [],
    q3: [],
    color: []
  }
}
```

And for the STOCSY:
```
function boxPlot(matrix) {
  {
    y: [],
    color: [], // as a string like "#ABCDEF"
  }
}
```
